pkg/chain/chain.go

```diff
@@ -1,164 +1,239 @@
 	OnKeepClosed(
-		keepAddress common.Address,
 		handler func(event *KeepClosedEvent),
 	) (subscription.EventSubscription, error)
 
 	// OnKeepTerminated installs a callback that will be called on terminating
 	// the given keep.
 	OnKeepTerminated(
-		keepAddress common.Address,
 		handler func(event *KeepTerminatedEvent),
 	) (subscription.EventSubscription, error)
 
 	// IsAwaitingSignature checks if the keep is waiting for a signature to be
 	// calculated for the given digest.
-	IsAwaitingSignature(keepAddress common.Address, digest [32]byte) (bool, error)
+	IsAwaitingSignature(digest [32]byte) (bool, error)
 
 	// IsActive checks if the keep with the given address is active and responds
 	// to signing request. This function returns false only for closed keeps.
-	IsActive(keepAddress common.Address) (bool, error)
+	IsActive() (bool, error)
 
 	// LatestDigest returns the latest digest requested to be signed.
-	LatestDigest(keepAddress common.Address) ([32]byte, error)
+	LatestDigest() ([32]byte, error)
 
 	// SignatureRequestedBlock returns block number from the moment when a
 	// signature was requested for the given digest from a keep.
 	// If a signature was not requested for the given digest, returns 0.
-	SignatureRequestedBlock(keepAddress common.Address, digest [32]byte) (uint64, error)
+	SignatureRequestedBlock(digest [32]byte) (uint64, error)
 
 	// GetPublicKey returns keep's public key. If there is no public key yet,
 	// an empty slice is returned.
-	GetPublicKey(keepAddress common.Address) ([]uint8, error)
+	GetPublicKey() ([]uint8, error)
 
 	// GetMembers returns keep's members.
-	GetMembers(keepAddress common.Address) ([]common.Address, error)
+	GetMembers() ([]KeepMemberID, error)
+
+	// IsThisOperatorMember returns true if the current operator belongs to the
+	// BondedECDSAKeep represented by this handle, false otherwise, or an error
+	// if the process of determining this fails.
+	IsThisOperatorMember() (bool, error)
+
+	// OperatorIndex returns the index of the current operator in this keep's
+	// set of members, or an error if the process of determining this fails. If
+	// the operator is not a member this will return -1 (and no error) and
+	// IsOperatorMember will return false.
+	OperatorIndex() (int, error)
 
 	// GetHonestThreshold returns keep's honest threshold.
-	GetHonestThreshold(keepAddress common.Address) (uint64, error)
+	GetHonestThreshold() (uint64, error)
 
 	// GetOpenedTimestamp returns timestamp when the keep was created.
-	GetOpenedTimestamp(keepAddress common.Address) (time.Time, error)
+	GetOpenedTimestamp() (time.Time, error)
 
 	// PastSignatureSubmittedEvents returns all signature submitted events
 	// for the given keep which occurred after the provided start block.
 	// All implementations should returns those events sorted by the
 	// block number in the ascending order.
 	PastSignatureSubmittedEvents(
-		keepAddress string,
+		// FIXME how can this be divorced from a block number?
```

Comment:
Maybe passing a flag describing the point of time we want to use, so e.g. keep creation or last redemption.

Reply:
I was actually thinking last night about using the event itself, so e.g. `sinceEvent Event`, where all the structs implement the `Event` interface. But… Due to how interfaces work in Go I'm not sure if there's a good way of doing this generically without re-exposing the block stuff. I haven't done a deep dive, but it feels like it could be a promising direction.